### DISCUSSION TITLE
Alpine doesn't work without initialized self/parent element

### DISCUSSION THREAD

**Discussion by tinyoverflow** [alpinejs/alpine#3218] [7. Bugs (archived)] [answered]
Hello,

I've noticed, that when trying to use Alpine events, it does not work unless the element itself or the parent has been initialized somehow. As far as I can tell, this behaviour is not documented.

<details>
  <summary>Example file</summary>
  
```html
<!doctype html>
<html>
    <head>
        <title>Test</title>
        <script defer src="https://unpkg.com/alpinejs@3.10.4/dist/cdn.min.js"></script>
    </head>
    <body x-init="console.log('init body')">
        <button @click="console.log('test')">Test</button>
        <button @click="console.log('test 2')" x-init="console.log('init 2')">Test 2</button>
    </body>
</html>
```
  
</details>

When removing the `x-init` attribute (also works with `x-data`) from the body tag, the button _Test_ stops working, while the button _Test 2_ still works, as it has the x-init attribute set. When removing this attribute from the button _Test 2_, it stops working, too.

Global events without a local data store or the x-init workaround won't work. The following example works unless the x-data attribute gets removed.

<details>
  <summary>Example file</summary>
  
```html
<!doctype html>
<html>
    <head>
        <title>Test</title>
        <script defer src="https://unpkg.com/alpinejs@3.10.4/dist/cdn.min.js"></script>
    </head>
    <body @show-alert="alert('Hello, World!')" x-data="{}">
        <button @click="$dispatch('show-alert')">Show Alert</button>
    </body>
</html>
```
  
</details>

**Comment by ekwoka** [comment]
It is well documented that directives need to be used nested within an x-data.

Otherwise they are not in a component at all and none of the directives should work.

https://alpinejs.dev/directives/data

**Reply by tinyoverflow** [reply]
I've read that page many times and I don't see a paragraph where it clearly states that wrapping something in an element, which has x-data set, is mandatory.

It also obviously isn't required, as it also works when I replace the empty x-data from the body with the x-init attribute. The x-init doesn't even have to evaluate anything for it to work.

**Reply by ekwoka** [reply]
Well, in every example it uses it and it even expresses how to get directives without data.

The x-init thing you're describing is actually a bug.

**Comment by SimoTod** [comment, marked as answer]
confirmed, it's not a bug. Alpine has always worked like that. It would be a big hit for the performance to let alpine crawl the whole page looking for x-attribute, especially on big pages. For that reason, Alpine is just looking for x-data first and it just crawls down starting from there. I agree that the documentation could have been clearer, but it's always been like that and most of the "good" tutorials you can find on the internet will tell you.

The docs itself is open source so feel free to improve it.

In regard to x-init, it didn't use to work in V2 but it's a concession Caleb made because a lot of people wanted it (Again, it's not well documented but it's the only exception: https://alpinejs.dev/directives/init#standalone-x-init). Basically adding just x-init creates a stateless component (equivalent to having an empty x-data + x-init).

**Reply by tinyoverflow** [reply]
Thank you really much for clarifying this! Every example uses the x-data attribute, because every example somehow relates to using data with it. I assumed that it's just the local data store, but that I don't need it if I don't have any data.

I'll check on how and where to explain this in the docs. The getting started and lifecycle pages would probably a good start. What do you think?
